owncloudApp/src/main/java/com/owncloud/android/providers/FileContentProvider.kt

```diff
@@ -0,0 +1,1470 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Bartek Przybylski
+ * @author David A. Velasco
+ * @author masensio
+ * @author David González Verdugo
+ * @author Christian Schabesberger
+ * Copyright (C) 2011  Bartek Przybylski
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.providers
+
+import android.accounts.AccountManager
+import android.content.*
+import android.database.Cursor
+import android.database.SQLException
+import android.database.sqlite.SQLiteDatabase
+import android.database.sqlite.SQLiteOpenHelper
+import android.database.sqlite.SQLiteQueryBuilder
+import android.net.Uri
+import android.os.CancellationSignal
+import android.os.ParcelFileDescriptor
+import android.text.TextUtils
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.SupportSQLiteQueryBuilder
+import com.owncloud.android.AppExecutors
+import com.owncloud.android.MainApp
+import com.owncloud.android.R
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.datamodel.UploadsStorageManager
+import com.owncloud.android.db.OwncloudDatabase
+import com.owncloud.android.db.ProviderMeta
+import com.owncloud.android.db.ProviderMeta.ProviderTableMeta
+import com.owncloud.android.lib.common.accounts.AccountUtils
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.shares.ShareType
+import com.owncloud.android.shares.datasources.OCLocalSharesDataSource
+import com.owncloud.android.shares.db.OCShare
+import com.owncloud.android.utils.FileStorageUtils
+import java.io.File
+import java.io.FileNotFoundException
+import java.util.*
+
+/**
+ * The ContentProvider for the ownCloud App.
+ */
+class FileContentProvider(val appExecutors: AppExecutors = AppExecutors()) : ContentProvider() {
+
+    private var mDbHelper: DataBaseHelper? = null
+
+    private var mUriMatcher: UriMatcher? = null
```

Comment:
the same as `mDbHelper`

And don't forget to remove !! ;-)

Reply:
Done